### PR TITLE
Allow rescue inside do/end blocks

### PR DESCRIPTION
**Pull request by presidentbeef**
Just copied https://github.com/ruby/ruby/commit/0ec889d7ed34f80b534dfc7a93bdd3175aba9ff9 although I'm not really sure why `brace_block` needs to change and `do_block` doesn't.

Addresses #262

### CODE REVIEW

**Comment by kyontan** [comment]
@zenspider I'm also affected with this problem, Could you review this PR?

**Comment by zenspider** [comment]
I fixed up the whitespacing a bit, but otherwise this looks great. Thanks!

**Comment by presidentbeef** [comment]
Thank you! 🙇

**Comment by craineum** [comment]
Is there a timeline on this getting into an official release? Thanks.

File: test/test_ruby_parser.rb [hunk 1]
```diff
@@ -3766,6 +3766,72 @@ def setup
 
     self.processor = RubyParser::V25.new
   end
+
+  def test_rescue_in_block
```
Comment (abrom) [inline test/test_ruby_parser.rb]: Should these tests be moved to the `TestRubyParserShared25Plus` module? Happy to create a PR if that's the case. Can't see why they wouldn't apply to 2.6+

**Pull request closed without merging**